src/app/cli/src/init/client.ml

```diff
@@ -1789,90 +1789,82 @@ let add_peers_graphql =
-                         accounts = Some []
-                       }
-                 }
-               |> Result.ok_or_failwith
-             in
-             Option.value ~default config_file
-           in
-           Genesis_ledger_helper.Config_loader.init_from_config_file_legacy
-             ~genesis_constants ~constraint_constants ~logger:(Logger.null ())
-             ~proof_level ~cli_proof_level:None ~genesis_dir runtime_config
-           >>| Or_error.ok_exn
+         let%bind m_conf =
+           Runtime_config.Json_loader.load_config_files ~conf_dir ~logger
+             config_file
+           >>| Or_error.ok
          in
-         let all_constants =
-           `Assoc
-             [ ( "genesis_state_timestamp"
-               , `String
-                   ( Block_time.to_time_exn
-                       consensus_constants.genesis_state_timestamp
-                   |> Core.Time.to_string_iso8601_basic ~zone:Core.Time.Zone.utc
-                   ) )
-             ; ("k", `Int (Unsigned.UInt32.to_int consensus_constants.k))
-             ; ( "coinbase"
-               , `String
-                   (Currency.Amount.to_mina_string
-                      precomputed_values.constraint_constants.coinbase_amount )
-               )
-             ; ( "block_window_duration_ms"
-               , `Int
-                   precomputed_values.constraint_constants
-                     .block_window_duration_ms )
-             ; ("delta", `Int (Unsigned.UInt32.to_int consensus_constants.delta))
-             ; ( "sub_windows_per_window"
-               , `Int
-                   (Unsigned.UInt32.to_int
-                      consensus_constants.sub_windows_per_window ) )
-             ; ( "slots_per_sub_window"
-               , `Int
-                   (Unsigned.UInt32.to_int
-                      consensus_constants.slots_per_sub_window ) )
-             ; ( "slots_per_window"
-               , `Int
-                   (Unsigned.UInt32.to_int consensus_constants.slots_per_window)
-               )
-             ; ( "slots_per_epoch"
-               , `Int
-                   (Unsigned.UInt32.to_int consensus_constants.slots_per_epoch)
-               )
-             ]
+         let default =
+           Runtime_config.of_json_layout
+             { Runtime_config.Json_layout.default with
+               ledger =
+                 Some
+                   { Runtime_config.Json_layout.Ledger.default with
+                     accounts = Some []
+                   }
+             }
+           |> Result.ok_or_failwith
          in
-         Core_kernel.printf "%s\n%!" (Yojson.Safe.to_string all_constants) ) )
+         let runtime_config = Option.value ~default m_conf in
+         let constants =
+           Runtime_config.Constants.load_constants' runtime_config
+         in
+         Genesis_ledger_helper.Config_loader.init_from_config_file ~genesis_dir
+           ~logger ~constants runtime_config
```

Comment:
This change is particularly irritating, changing `Logger.null` to `logger` (non-null above).